### PR TITLE
NEXT-23682 - Add varnish xkey support

### PR DESCRIPTION
**Pull request by shyim**

### CODE REVIEW

**Review by keulinho — APPROVED** [review]

File: guides/hosting/infrastructure/reverse-http-cache.md [hunk 1]
```diff
@@ -456,6 +456,235 @@ sub soft_purge_page {
...
+
+    # Handle PURGE
+    if (req.method == "PURGE") {
+        if (client.ip !~ purgers) {
+            return (synth(403, "Forbidden"));
+        }
+        if (req.http.xkey) {
+            set req.http.n-gone = xkey.purge(req.http.xkey);
+
+            # Replace this line for
```
Comment (Christian-Rades) [inline guides/hosting/infrastructure/reverse-http-cache.md:530]: Replace for what?

File: guides/hosting/infrastructure/reverse-http-cache.md [hunk 2]
```diff
@@ -456,6 +456,235 @@ sub soft_purge_page {
...
+    "127.0.0.1";
+    "localhost";
+    "::1";
+}
+
+sub vcl_recv {
+    # Mitigate httpoxy application vulnerability, see: https://httpoxy.org/
+    unset req.http.Proxy;
+
+    # Strip query strings only needed by browser javascript. Customize to used tags.
```
Comment (Christian-Rades) [inline guides/hosting/infrastructure/reverse-http-cache.md:505]:
```suggestion
    #  Ignore query strings that are only necessary for the js on the client.  Customize as needed.
```

File: guides/hosting/infrastructure/reverse-http-cache.md [hunk 3]
```diff
@@ -456,6 +456,235 @@ sub soft_purge_page {
...
+
+Varnish Config:
+
+```vcl
+vcl 4.0;
+
+import std;
+import xkey;
+
+# You should specify here all your app nodes and use round robin to select a backend
```
Comment (Christian-Rades) [inline guides/hosting/infrastructure/reverse-http-cache.md:488]:
```suggestion
# Specify your app nodes here. Use round-robin balancing to add more than one.
```

**Review by Christian-Rades — CHANGES_REQUESTED** [review]

File: guides/hosting/infrastructure/reverse-http-cache.md [hunk 4]
```diff
@@ -456,6 +456,235 @@ sub soft_purge_page {
...
+
+    # Handle PURGE
+    if (req.method == "PURGE") {
+        if (client.ip !~ purgers) {
+            return (synth(403, "Forbidden"));
+        }
+        if (req.http.xkey) {
+            set req.http.n-gone = xkey.purge(req.http.xkey);
+
+            # Replace this line for
```
Comment (shyim) [inline guides/hosting/infrastructure/reverse-http-cache.md:530]:
```suggestion
            # To enable soft-purge replace this line with above
```

**Review by Christian-Rades — APPROVED** [review]

**Pull request merged**